content/components/banner.mdx

```diff
@@ -8,15 +8,254 @@ tags:
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="critical banner"
+    src="https://github.com/primer/design/assets/813754/8b135c8e-898d-423c-81c4-9dbb4736a4d5"
+  />
+  <Box as="p" mt="0">
+    <h4>Critical</h4>
+    Use critical banners to signal critical errors, system failures, or necessary destructive actions. Examples: form validation error summary, repo deletion confirmations.
+  </Box>
+</Box>
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="warning banner"
+    src="https://github.com/primer/design/assets/813754/fc82519b-60e1-45b3-a0de-c5f89108e96e"
+  />
+  <Box as="p" mt="0">
+    <h4>Warning</h4>
+    Use warning banners to alert users about potential issues or impactful changes. For example possible connectivity concerns or hidden items due to active filters.
+  </Box>
+</Box>
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="success banner"
+    src="https://github.com/primer/design/assets/813754/fe04a355-d343-4a79-9948-efc64d6de76e"
+  />
+  <Box as="p" mt="0">
+    <h4>Success</h4>
+    Use success banners to indicate task completion or successful actions. For example to confirm saving of settings or successful issue transfer. Only use success messages if the success is not easily apparent in the UI.
+  </Box>
+</Box>
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="info banner"
+    src="https://github.com/primer/design/assets/813754/172e3eb4-35ea-4d75-a015-6d00d3ebd914"
+  />
+  <Box as="p" mt="0">
+    <h4>Info</h4>
+    Use info banners to inform about non-critical context and information. For example feature prompts or ongoing processes.
+  </Box>
+</Box>
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="upsell banner"
+    src="https://github.com/primer/design/assets/813754/48b37115-c0e0-40c3-83f1-3e831178f175"
+  />
+  <Box as="p" mt="0">
+    <h4>Upsell</h4>
+    Use upsell banners to inform the users about a feature that can be enabled by upgrading their plan.
+  </Box>
+</Box>
+
+### Title
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="banner without title and banner with title"
+    src="https://github.com/primer/design/assets/813754/c5876803-7b5b-486d-b2ec-27fcfdf0c5d8"
+  />
+  <Box as="p" mt="0">
+    <h4>Title</h4>
+    Use a title to consisly present the topic of the banner when the message of the banner is complex and too long to scan.
```

Comment:
Can you add something about title being required but may be visually hidden?

Reply:
@joshblack is this how we implement it, or is there a separate a11y title which will just use the title the user defines if they do?

Reply:
I am not sure if this is the right place for a discussion, but I am a bit afraid that if the title is not hidden by default, we will get a lot of duplicates with title and body being the same text.

My only thought to improve this would be to have two props like `aria-title` and `title` where `aria-title` is set to `title` if present. So, using typescript we could require on of the two to be set.

But I am no react dev and this may be an uncommon or anti-pattern.